### PR TITLE
joint_state_broadcaster to use realtime tools

### PR DESCRIPTION
**Pull request by bmagyar**

### CODE REVIEW

File: joint_state_broadcaster/package.xml
Comment (destogl) [inline joint_state_broadcaster/package.xml:9]: Let's change list of maintainers in a separate PR and in all packages at once.

File: joint_state_broadcaster/src/joint_state_broadcaster.cpp
Comment (destogl) [inline joint_state_broadcaster/src/joint_state_broadcaster.cpp:343]: You forgot to lock the variables.

File: joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
Comment (destogl) [inline joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp:99]: Not sure about 'realtime' prefix though...

I would call the normal one with 'pub_' suffix and this one 'publisher_'


One way or another we should sync naming in all broadcasters we have.

**Review by destogl — CHANGES_REQUESTED** [review]

File: joint_state_broadcaster/package.xml
Comment (bmagyar) [reply joint_state_broadcaster/package.xml:9]: Yessir

File: joint_state_broadcaster/src/joint_state_broadcaster.cpp
Comment (bmagyar) [reply joint_state_broadcaster/src/joint_state_broadcaster.cpp:343]: Oh no! You are right

File: joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
Comment (bmagyar) [reply joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp:99]: I didn't have a better idea... I actually want to discuss the realtime publisher API, I'm r ally.not happy with it

File: joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
Comment (destogl) [inline joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp:105]: It is really needed to set the initial value to `nullptr`?

If so, we should add this to all controllers (`good-first-issue`). If needed please open follow-up issue.

File: joint_state_broadcaster/src/joint_state_broadcaster.cpp
Comment (destogl) [inline joint_state_broadcaster/src/joint_state_broadcaster.cpp:355]: It looks like you mixed the publishers?

```suggestion
  if (realtime_joint_state_publisher_ && realtime_joint_state_publisher_->trylock())
```

**Review by destogl — APPROVED** [review]
It looks good, but please check the comment about mixed publishers!

Thanks!

File: joint_state_broadcaster/src/joint_state_broadcaster.cpp
Comment (bmagyar) [reply joint_state_broadcaster/src/joint_state_broadcaster.cpp:355]: nice catch, thanks! :D

File: joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
Comment (bmagyar) [reply joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp:105]:
```suggestion
    realtime_joint_state_publisher_;
```
du has recht, not strictly needed

File: joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
Comment (bmagyar) [inline joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp:113]:
```suggestion
    realtime_dynamic_joint_state_publisher_;
```

**Comment by destogl** [comment]
It seems you have to also adjust tests too because you deleted an internal variable for storing messages.

**Review by destogl — APPROVED** [review]
Looking good. Let's just wait for the CI.

**Pull request merged**
